File: so-service/src/main/java/com/ss/smartoffice/soservice/seed/dayrange/DayRangeService.java
package com.ss.smartoffice.soservice.seed.dayrange;

import java.util.Optional;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.context.annotation.Scope;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PatchMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.ss.smartoffice.shared.common.CommonUtils;
import com.ss.smartoffice.shared.model.SmartOfficeException;



@Controller
@RestController
@RequestMapping(value="/seed/day-range")
@Scope("prototype")
public class DayRangeService {
@Autowired
 DayRangeRepository dayrangerepository;
@Autowired
 CommonUtils commonutils;
@GetMapping
public Iterable<DayRange> getDayRanges() throws SmartOfficeException{
	return dayrangerepository.findAll();
}
@GetMapping(value="/{id}")
public Optional<DayRange> getDayRangeById(@PathVariable(value="id")Integer id) throws SmartOfficeException {
	
	return dayrangerepository.findById(id);
}
@PostMapping
public DayRange addDayRange(@RequestBody DayRange dayrange)throws SmartOfficeException{
	dayrange.setCreatedBy(commonutils.getLoggedinEmployeeId());
	dayrange.setIsEnabled("Y");
    return dayrangerepository.save(dayrange);
}
@PatchMapping("/{id}/update")
public DayRange  addOrUpdateLines(@RequestBody DayRange dayrange,@PathVariable(value = "id") Integer id)throws SmartOfficeException{
	dayrange.setModifiedBy(commonutils.getLoggedinEmployeeId());
	dayrange.setIsEnabled("Y");
	return dayrangerepository.save(dayrange);
}
@DeleteMapping("/{id}/delete")
public void deleteDayRange(@PathVariable(value="id") Integer id) throws SmartOfficeException{
	    dayrangerepository.deleteById(id);
}


}

File: so-ui/src/app/pages/operation/day-range/day-range.service.ts
import { Injectable } from '@angular/core';
import { environment } from './../../../../environments/environment';
import { Http, Response } from '@angular/http';
import { CommonService } from '../../../shared/common/common.service';
import {DayRange} from './vo/day-range';

@Injectable()
export class  DayRangeService {
    baseUrl:string=environment.serviceApiUrl;
    xapikey:string=environment.xapikey;
    clientId:string=environment.clientId;

    constructor(private http:Http,private commonService:CommonService) { }

    getDayRanges(){
        return this.http.get(this.baseUrl+'seed/day-range',this.commonService.jwt()).map((response:Response)=>response.json());
    }
    getDayRangeById(id:string){
        return this.http.get(this.baseUrl+'seed/day-range/'+id,this.commonService.jwt()).map((response:Response)=>response.json());
    }

    addDayRange(dayRange:DayRange){
        return this.http.post(this.baseUrl+'seed/day-range',dayRange,this.commonService.jwt()).map((response:Response)=>response.json());
    }
    deleteDayRange(id:string){
        return this.http.delete(this.baseUrl+'seed/day-range/'+id+'/delete',this.commonService.jwt()).map((response:Response)=>response);
    }
    addOrUpdateLines(dayrange:DayRange,id:string){
        return this.http.patch(this.baseUrl+'seed/day-range/'+id+'/update',dayrange,this.commonService.jwt()).map((response:Response)=>response.json());
    }
}